tests/bluetooth/bsim/compile.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
```

Comment:
Since we are restructuring the bsim directory, would it make sense to also split this file into smaller compile files?

Reply:
Would be nice. Then I don't need to comment out other test harnesses if I want to compile only the mesh ones.

Reply:
The think that would make the most sense would be switching to twister for building and running bsim tests. Then we could do away with the ad-hoc shell scripts that we have.
But I thought that'd be too much scope, hence the rename-only PR.

Reply:
This will become much nicer once we get the pytest plugin and twister v2 (ETA ??), then we can run and filter tests properly.

Reply:
Looks good to me